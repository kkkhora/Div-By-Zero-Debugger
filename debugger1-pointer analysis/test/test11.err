Pointer Analysis Results:
  %i      : { @(%i = alloca i32, align 4); }
  %sum    : { @(%sum = alloca i32, align 4); }
  %z      : { @(%z = alloca i32, align 4); }

get store value operand: i32 0
get store value operand: i32 0
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: Zero     
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: Zero     
get store value operand:   %add = add nsw i32 %1, 1
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: Zero     
get store value operand:   %inc = add nsw i32 %2, 1
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: Zero     
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: Zero     
get store value operand:   %div = sdiv i32 %3, %4
get store value operand: i32 0
get store value operand: i32 0
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %add = add nsw i32 %1, 1
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %inc = add nsw i32 %2, 1
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %div = sdiv i32 %3, %4
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %add = add nsw i32 %1, 1
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %inc = add nsw i32 %2, 1
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %div = sdiv i32 %3, %4
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %add = add nsw i32 %1, 1
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
Load type is integer
  %i = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
Load type is integer
  %sum = alloca i32, align 4
Load in InMem
Load Domain: MaybeZero
get store value operand:   %div = sdiv i32 %3, %4
Dataflow Analysis Results:
Instruction:   %i = alloca i32, align 4
In set: 

Out set: 


Instruction:   %sum = alloca i32, align 4
In set: 

Out set: 


Instruction:   %z = alloca i32, align 4
In set: 

Out set: 


Instruction:   store i32 0, i32* %sum, align 4
In set: 

Out set: 
    [ %sum     |-> Zero      ]

Instruction:   store i32 0, i32* %i, align 4
In set: 
    [ %sum     |-> Zero      ]
Out set: 
    [ %i       |-> Zero      ]
    [ %sum     |-> Zero      ]

Instruction:   br label %for.cond
In set: 
    [ %i       |-> Zero      ]
    [ %sum     |-> Zero      ]
Out set: 
    [ %i       |-> Zero      ]
    [ %sum     |-> Zero      ]

Instruction:   %0 = load i32, i32* %i, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %cmp = icmp slt i32 %0, 10
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   br i1 %cmp, label %for.body, label %for.end
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %1 = load i32, i32* %sum, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %add = add nsw i32 %1, 1
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   store i32 %add, i32* %sum, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   br label %for.inc
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %2 = load i32, i32* %i, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %inc = add nsw i32 %2, 1
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   store i32 %inc, i32* %i, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   br label %for.cond
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %3 = load i32, i32* %i, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %4 = load i32, i32* %sum, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %4       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   %div = sdiv i32 %3, %4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %4       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %4       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %div     |-> Uninit    ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]

Instruction:   store i32 %div, i32* %z, align 4
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %4       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %div     |-> Uninit    ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %4       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %div     |-> Uninit    ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
    [ %z       |-> Uninit    ]

Instruction:   ret void
In set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %4       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %div     |-> Uninit    ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
    [ %z       |-> Uninit    ]
Out set: 
    [ %0       |-> MaybeZero ]
    [ %1       |-> MaybeZero ]
    [ %2       |-> MaybeZero ]
    [ %3       |-> MaybeZero ]
    [ %4       |-> MaybeZero ]
    [ %add     |-> MaybeZero ]
    [ %cmp     |-> MaybeZero ]
    [ %div     |-> Uninit    ]
    [ %i       |-> MaybeZero ]
    [ %inc     |-> MaybeZero ]
    [ %sum     |-> MaybeZero ]
    [ %z       |-> Uninit    ]

