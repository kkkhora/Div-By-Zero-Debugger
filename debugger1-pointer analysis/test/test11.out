Running DivZero on f
Potential Instructions by DivZero: 
  %div = sdiv i32 %3, %4
